homeassistant/components/netatmo/config_flow.py

```diff
@@ -45,11 +65,113 @@ def extra_authorize_data(self) -> dict:
+            }
+            self.options.update(user_input)
+            if new_client:
+                return await self.async_step_public_weather(
+                    user_input={CONF_NEW_AREA: new_client}
+                )
+
+            return await self._update_options()
+
+        weather_areas = list(self.options[CONF_WEATHER_AREAS])
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_WEATHER_AREAS, default=weather_areas,
+                ): cv.multi_select(weather_areas),
+                vol.Optional(CONF_NEW_AREA): str,
+            }
+        )
+        return self.async_show_form(
+            step_id="public_weather_areas", data_schema=data_schema, errors=errors,
+        )
+
+    async def async_step_public_weather(self, user_input=None):
+        """Manage configuration of Netatmo public weather sensors."""
+        if user_input is not None and CONF_NEW_AREA not in user_input:
+            self.options[CONF_WEATHER_AREAS][user_input[CONF_AREA_NAME]] = user_input
+            return await self.async_step_public_weather_areas()
+
+        orig_options = self.config_entry.options.get(CONF_WEATHER_AREAS, {}).get(
+            user_input[CONF_NEW_AREA], {}
+        )
+
+        default_longitude = self.hass.config.longitude
+        default_latitude = self.hass.config.latitude
+        default_size = 0.04
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(CONF_AREA_NAME, default=user_input[CONF_NEW_AREA]): str,
+                vol.Optional(
+                    CONF_LAT_NE,
+                    default=orig_options.get(
+                        CONF_LAT_NE, default_latitude + default_size
+                    ),
+                ): cv.latitude,
+                vol.Optional(
+                    CONF_LON_NE,
+                    default=orig_options.get(
+                        CONF_LON_NE, default_longitude + default_size
+                    ),
+                ): cv.longitude,
+                vol.Optional(
+                    CONF_LAT_SW,
+                    default=orig_options.get(
+                        CONF_LAT_SW, default_latitude - default_size
+                    ),
+                ): cv.latitude,
+                vol.Optional(
+                    CONF_LON_SW,
+                    default=orig_options.get(
+                        CONF_LON_SW, default_longitude - default_size
+                    ),
+                ): cv.longitude,
+                vol.Required(
+                    CONF_PUBLIC_MODE, default=orig_options.get(CONF_PUBLIC_MODE, "avg"),
+                ): vol.In(["avg", "max"]),
+                vol.Required(
+                    CONF_SHOW_ON_MAP, default=orig_options.get(CONF_SHOW_ON_MAP, False),
+                ): bool,
+            }
+        )
+
+        return self.async_show_form(step_id="public_weather", data_schema=data_schema)
+
+    async def _update_options(self):
```

Comment:
This helper method doesn't need to be  a coroutine since we don't await inside.

Reply:
Actually we do in L105.

Reply:
There's no await inside this method. That's what matters. We of course need to await the method currently since we have defined it as a coroutine function.

Reply:
Oh, true. Fixed that in https://github.com/home-assistant/core/pull/35571/commits/5786afbd3df58eedf0155ea104f9f7a34c93807d
Thanks for pointing that out.